pkg/registry/apis/dashboards/access/sql_dashboards.go

```diff
@@ -0,0 +1,384 @@
+	}
+}
+
+const selector = `SELECT 
+	dashboard.org_id, dashboard.id,
+	dashboard.uid,slug,
+	dashboard.folder_uid,
+	dashboard.created,dashboard.created_by,CreatedUSER.login,
+	dashboard.updated,dashboard.updated_by,UpdatedUSER.login,
+	plugin_id,
+	dashboard_provisioning.name as origin_name,
+	dashboard_provisioning.external_id as origin_path,
+	dashboard_provisioning.check_sum as origin_key,
+	dashboard_provisioning.updated as origin_ts,
+	dashboard.version,
+	title,
+	dashboard.data
+  FROM dashboard 
+  LEFT OUTER JOIN dashboard_provisioning ON dashboard.id = dashboard_provisioning.dashboard_id
+  LEFT OUTER JOIN user AS CreatedUSER ON dashboard.created_by = CreatedUSER.id 
+  LEFT OUTER JOIN user AS UpdatedUSER ON dashboard.created_by = UpdatedUSER.id 
+  WHERE is_folder = false`
+
+// GetDashboard implements DashboardAccess.
+func (a *dashboardSqlAccess) GetDashboard(ctx context.Context, orgId int64, uid string) (*DashboardRow, error) {
+	rows, err := a.doQuery(ctx, selector+`
+		AND dashboard.org_id=$1
+		AND dashboard.uid=$2`, orgId, uid)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() { _ = rows.Close() }()
+
+	row, err := rows.Next()
+	if err != nil {
+		return nil, err
+	}
+	if row == nil {
+		// NOTE, this may be access control!
+		return nil, k8serrors.NewNotFound(v0alpha1.DashboardResourceInfo.GroupResource(), uid)
+	}
+	return row, nil
+}
+
+// GetDashboards implements DashboardAccess.
+func (a *dashboardSqlAccess) GetDashboards(ctx context.Context, orgId int64, continueToken string, skipBody bool) (DashboardRows, error) {
+	token, err := readContinueToken(continueToken)
+	if err != nil {
+		return nil, err
+	}
+	rows, err := a.doQuery(ctx, selector+`
+		AND dashboard.org_id=$1
+		AND dashboard.created >= $2
+		ORDER BY dashboard.org_id asc,dashboard.updated asc	
+	`, orgId, time.UnixMilli(token.updated))
+	if err == nil {
+		rows.advanceTo(token)
+	}
+	return rows, nil // the caller must Close!
+}
+
+func (a *dashboardSqlAccess) doQuery(ctx context.Context, query string, args ...any) (*rowsWrapper, error) {
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+	rows, err := a.sess.Query(ctx, query, args...)
+	return &rowsWrapper{
+		rows: rows,
+		a:    a,
+		// This looks up rules from the permissions on a user
+		canReadDashboard: accesscontrol.Checker(user, dashboards.ActionDashboardsRead),
+	}, err
+}
+
+type rowsWrapper struct {
+	a     *dashboardSqlAccess
+	rows  *sql.Rows
+	idx   int
+	total int64
+
+	canReadDashboard func(scopes ...string) bool
+
+	pending *DashboardRow
+}
+
+func (r *rowsWrapper) advanceTo(token continueToken) {
+	if token.uid != "" {
+		for {
+			row, err := r.Next()
+			if row == nil || err != nil {
+				return // ??
+			}
+			if row.Dash.Name == token.uid || row.UpdatedTime > token.updated {
+				r.pending = row
+				return
+			}
+		}
+	}
+}
+
+func (r *rowsWrapper) Close() error {
+	return r.rows.Close()
+}
+
+func (r *rowsWrapper) Next() (*DashboardRow, error) {
```

Comment:
@DanCech -- this is the logic/structure that needs the most attention/review

This is returning a wrapped row set that you can keep reading until things are too big.  I'm not sure how well this really helps us or if there are better ways to accomplish the same thing.

Functionally, we want to read dashboards -- but stop after hitting max size 🤔 

maybe better to just pass the size into the request and return an array 🤔